src/qt/aboutdialog.cpp

```diff
@@ -30,3 +38,28 @@ void AboutDialog::on_buttonBox_accepted()
 {
     close();
 }
+
+void AboutDialog::handlePressVersionInfoButton()
+{
+    std::string client_message_out;
+    std::string change_log;
+    GRC::Upgrade::UpgradeType upgrade_type = GRC::Upgrade::UpgradeType::Unknown;
+
+
+    GRC::Upgrade::CheckForLatestUpdate(client_message_out, change_log, upgrade_type, false, false);
```

Comment:
This completely locks the GUI thread for a few seconds if you have bad connectivity.

Reply:
I actually tested this with no connectivity, and it was faster than I could get to the ok. It is not worth the hassle to do the threadsafe pattern on this, although we could.

Reply:
Do you really think it is worth the plumbing?

Reply:
Eh, not really on second thought.